mod_reforged/hooks/entity/world/locations/legendary/black_monolith_location.nut

```diff
@@ -42,5 +46,59 @@
 	q.onBeforeCombatStarted = @() function()
 	{
 		this.location.onBeforeCombatStarted();
+
+		// Vanilla adds up to 6 new troops before every combat up to a maximum of 47 total troops
+		// We emulate that behavior using a custom entity pool
+
+		if (this.m.Troops.len() == this.m.NumTroopsMax || this.m.NumTroopsToRecoverBeforeCombat == 0)
+			return;
+
+		local pool = this.getTroopRecoveryPool();
+		if (pool.len() == 0)
+			return;
+
+		for (local i = 0; i < this.m.NumTroopsToRecoverBeforeCombat && this.m.Troops.len() <= this.m.NumTroopsMax; i++)
+		{
+			::Const.World.Common.addTroop(this, { Type = pool.roll() }, false);
+		}
+	}
+
+	// Reforged added function in this class
+	q.getTroopRecoveryPool <- function()
+	{
+		local pool = ::MSU.Class.WeightedContainer().addMany(1, [
+			::Const.World.Spawn.Troops.SkeletonMedium,
+			::Const.World.Spawn.Troops.SkeletonMediumPolearm,
+			::Const.World.Spawn.Troops.RF_SkeletonMediumElite,
+			::Const.World.Spawn.Troops.RF_SkeletonMediumElitePolearm,
+			::Const.World.Spawn.Troops.RF_SkeletonHeavyLesser,
+			::Const.World.Spawn.Troops.SkeletonHeavy,
+			::Const.World.Spawn.Troops.SkeletonPriest,
+			::Const.World.Spawn.Troops.Vampire
+		]);
+
+		local function addToPoolIfFewerThan( _troopType, _count )
+		{
+			local numPresent = 0;
+			foreach (t in this.getTroops())
+			{
+				if (t.Script == _troopType.Script)
+				{
+					numPresent++;
+				}
+			}
+
+			if (numPresent < _count)
+			{
+				pool.add(_troopType, _count - numPresent);
+			}
+		}
+
+		addToPoolIfFewerThan(::Const.World.Spawn.Troops.RF_SkeletonDecanus, 3);
+		addToPoolIfFewerThan(::Const.World.Spawn.Troops.RF_SkeletonCenturion, 2);
+		addToPoolIfFewerThan(::Const.World.Spawn.Troops.RF_SkeletonLegatus, 1);
+		addToPoolIfFewerThan(::Const.World.Spawn.Troops.RF_VampireLord, 1);
```

Comment:
This can currently cause more than the desired number of these units to be recovered because they end up in the pool and the pool rolls randomly. There needs to be a way to prevent them from exceeding that number or we simply decide that we don't recover these entities.

Reply:
I've pushed a commit that addresses this - it's a bit hardcoded and I don't like it 100% but it should suffice for now.